Working on calculating desired gain for mic array

aditya-uw opened the pull request:
Initial commits of notebooks to work on this problem:
`20240621__modelling_bat_source_pressure_respect_to_microphone.ipynb`: 
- Here is where I have the initial code to plot a scenario to find the intensity of a bat call emitted at different positions away from a microphone assuming an SPL, mouth radius, and bat frequency.
- I am also using the piston in an infinite baffle to model the bat's directivity.

Will be working on this some more to make it clear and understandable; next steps include:
- Adding a diagram to show my setup
- Plotting the directivity I am using
- Clarifying if I am doing a bat-centric model or a microphone-centric model (I think I keep getting confused myself)

aditya-uw committed:
added notebooks for modeling bat sound field to microphone

aditya-uw committed:
added diagram directivity and ran notebook

aditya-uw committed:
added notes of bat-centric and mic-centric models

aditya-uw commented:
@leewujung: I made some changes:
- I added a diagram of my setup in `20240622__modelling_bat_source_pressure_respect_to_microphone.ipynb`. 
- The 45 degree lines you were seeing were because of the directivity of my bat-centric model. The frequency I was using was 100kHz so there were dips when I computed my directivity using the piston in infinite baffle model.
![image](https://github.com/uw-echospace/duty-cycle-investigation/assets/46602344/e368ea40-8aae-47a1-904a-ea57ee454bd6)
- I reduced my frequency to 60kHz just because that's closer to our typical peak frequency.
- I realize that this is a bat-centric setup so I will work on a microphone-centric model. I have uploaded my notes on what I think the differences are in `20240622__bat_mic_model.pdf`. And what I think I need for the mic-centric model.

Let me know your thoughts!

aditya-uw commented:
Oh my model setup does not seem to show up in the notebook `20240622__modelling_bat_source_pressure_respect_to_microphone.ipynb`, you can find it as `20240622__model_setup.png` in the same folder `daily_notebooks/`

leewujung commented:
Maybe you fix it? it's good to have everything in one place.

aditya-uw committed:
trying to fix image rendering in notebook on GitHub

aditya-uw committed:
updated model setup picture

aditya-uw committed:
updated model setup picture again

aditya-uw committed:
set frequency to 60kHz and added in directivity plot again

aditya-uw committed:
added README.md and updated .gitignore

aditya-uw committed:
removed foliage dets.csv

aditya-uw committed:
added comments to bat-centric model notebook

aditya-uw committed:
added work testing several other mics

aditya-uw committed:
renamed file to represent current work

aditya-uw commented:
@leewujung I just made a new notebook where I plugged in sensitivity and gain settings for some of the mics you mentioned in Slack. I first calculate the SPL of any microphone and then use each microphone's sensitivity and gain to get the voltage. The order of microphones are:
1) Audiomoth: Sensitivity at -38dBV/Pa with gain at -5.7dB at medium gain with respect to high at 0dB (Kitzes lab report)
2) Echometer Touch 2 Pro: Sensitivity at -38dBV/Pa with gain at 0dB setting (User guide on Wildlife Acoustics)
3) SONG METER SM4BAT FS: Sensitivity at -47dBV re 1Pa with default gain at 12dB (User guide on Wildlife Acoustics)
4) Avisoft-Bioacoustics CM24/CMPA: Sensitivity at 500mV/Pa with uncertain gain settings (Did not find so set as 0dB)
5) FG-23629-D65 Momimic: Sensitivity at -55dBV/0.1Pa with gain at 40dB. (Sensitivity graph + Email)

Overall trends:
1) Audiomoth, Echometer Touch 2 Pro, Song Meter, and Momimic seem to range from 0.2mV to 80mV for SLs from 110dB to 150dB re 20uPa. 
2) The Avisoft reports a range of 28mV to 3V for the same range of SLs so it is indeed more sensitive.

aditya-uw committed:
updated notebook

leewujung commented on daily_notebook/20240701__modelling_bat_source_pressure_respect_to_microphone.ipynb:
```diff
@@ -0,0 +1,727 @@
+{
```
<p>We should do both of the frequency groups?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/uw-echospace/duty-cycle-investigation/pull/4/discussion/#comment-1663203661'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

leewujung replied:
<p>I think directionality should not be included as part of transmission loss, so best to remove it from this plot.</p><p> </p><p>BUT, directionality is important so I think you can make a separate plot for that. It is useful to be able to read the approx values directly from a figure.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/uw-echospace/duty-cycle-investigation/pull/4/discussion/#comment-1663203662'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

leewujung replied:
<p>You should make the below a function so that you are not typing in constant repeatedly. That is prone to error.</p>

<pre class="ql-syntax" spellcheck="false">mic_sensitivity_dBVperPa = -38
gain_setting_in_dB = -5.7
dBperPa = 94
dist = 20.0
RLper20uPa_at_dist = RL_dB[:,np.where(np.isclose(r,dist,atol=0.01))[0]]
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/uw-echospace/duty-cycle-investigation/pull/4/discussion/#comment-1663203663'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

leewujung commented:
@aditya-uw : I added some comments via ReviewNB above.

Overall I think you could keep the directionality and transmission loss plot that you had before (the 2D color plot -- one axis is distance, the other axis is angular direction). You can have one version for each frequency. This way you could add SL to that and get the SPL easily.

And then you can make a function so that you can convert from SPL to voltage easily by plugging the sensitivity and gain -- Remember modularity is your friend! :) 

Do you think the numbers you get are reasonable given what the Audiomoth recording had received?

aditya-uw committed:
added in new work reviewing audiomoth recordings and adding frequencies to bat model

aditya-uw commented:
@leewujung: I added in 2 notebooks `20240702...` under `daily_notebooks/`
1) Under `20240702__examining_and_reviewing_call_in_audiomoth_recordings.ipynb` I opened up some representative calls as a waveform and spectrogram for each location to examine the voltages received. 
   - Regardless of frequency, the range we received the detected calls are from 4mV to more than 150mV. 
   - Most of the time the nearby noise is around 4mV as well so I assume the calls less than 4mV were not detected by our detector. 
   - The high values of more than 150mV are seen for a few low-frequency and high-frequency calls.

I wonder if the high-frequency bats are raising their SL or are closer in comparison to the low-frequency bats. Almost all of their calls are around the same level as the low-frequency calls.

2) Under `20240702__modelling_bat_source_pressure_respect_to_microphone.ipynb`, I added in 3 separate frequency groups (25kHz, 40kHz, and 55kHz) and added in a distance array to play around with separate distances for each group.
   - To hear the high frequency bats (55kHz) of Telephone Field from 20m, the Audiomoth voltage range would actually be from 0.005mV to 0.5mV. So we would need much more amplification than I previously reported because I was only looking at low-frequency bats.

aditya-uw committed:
updated notebooks to remove unneccessary things

aditya-uw committed:
renamed function name
